### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by aatuny**

### CODE REVIEW

File: .github/workflows/kb-node-tests.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    - run: npm test
+    - run: npm run build --if-present
+
+  license-scan:
+    name: License compliance check
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: mikaelvesavuori/license-compliance-action@main
```
Comment (natlibfi-jonollil) [inline .github/workflows/kb-node-tests.yml:37]: `     - uses: mikaelvesavuori/license-compliance-action@1`
Use major version number reference when possible

File: package.json [hunk 2]
```diff
@@ -13,63 +13,69 @@
 		"type": "git",
 		"url": "git@github.com:natlibfi/melinda-record-import-transformer-publication-archives.git"
 	},
-	"license": "MIT",
-	"version": "2.0.9",
+	"license": "UNLICENSED",
```
Comment (natlibfi-jonollil) [inline package.json:16]: Why unlicensed and no MIT?

File: src/transform/convert/index.js [hunk 3]
```diff
@@ -1,7 +1,74 @@
-import convertRecord from './common';
+import {MarcRecord} from '@natlibfi/marc-record';
+import moment from 'moment';
```
Comment (natlibfi-jonollil) [inline src/transform/convert/index.js:2]: Avoid using moment. Moment is deprecated https://momentjs.com/docs/#/-project-status/

Comment (natlibfi-jonollil) [reply src/transform/convert/index.js:2]: https://date-fns.org/ is okayish

**Review by natlibfi-jonollil — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (aatuny) [reply package.json]: Changed

File: src/transform/convert/index.js (hunk 3 above)
Comment (aatuny) [reply src/transform/convert/index.js:2]: Migration from moment to non-deprecated dependency is done in unified manner for all existing conversions

**Review by ssairanen — CHANGES_REQUESTED** [review]
If text data in xml file has either **&amp;** or **&quot;** following error messages (respectively) will be shown and transformation crashes (also other xml entities &lt, &gt, &apos; were tested, no problems there):

```
/.../melinda-record-import-transformer-publication-archives/node_modules/sax/lib/sax.js:658
    er = new Error(er)
         ^

Error: Invalid character in entity name
Line: 11
Column: 1408
Char:  
    at error (/.../melinda-record-import-transformer-publication-archives/node_modules/sax/lib/sax.js:658:10
...

/.../melinda-record-import-transformer-publication-archives/node_modules/sax/lib/sax.js:658
    er = new Error(er)
         ^

Error: No whitespace between attributes
Line: 11
Column: 115
Char: o
    at error (/.../melinda-record-import-transformer-publication-archives/node_modules/sax/lib/sax.js:658:10)
...

**Comment by aatuny** [comment]
> If text data in xml file has either `&amp;` or `&quot;` following error messages (respectively) will be shown and transformation crashes (also other xml entities `&lt;`, `&gt;`, `&apos;` were tested, no problems there):
> 
> ```
> /.../melinda-record-import-transformer-publication-archives/node_modules/sax/lib/sax.js:658
>     er = new Error(er)
>          ^
> 
> Error: Invalid character in entity name
> Line: 11
> Column: 1408
> Char:  
>     at error (/.../melinda-record-import-transformer-publication-archives/node_modules/sax/lib/sax.js:658:10
> ...
> 
> /.../melinda-record-import-transformer-publication-archives/node_modules/sax/lib/sax.js:658
>     er = new Error(er)
>          ^
> 
> Error: No whitespace between attributes
> Line: 11
> Column: 115
> Char: o
>     at error (/.../melinda-record-import-transformer-publication-archives/node_modules/sax/lib/sax.js:658:10)
> ...
> ```

Good catch! Parser has now been improved. Only tag that is not allowed and results into error is &lt; since that would be interpreted as new XML element by the parsers. Integration test now cover other entities.

**Pull request merged**